File: src/main/java/com/skocur/imagecipher/tools/ColorParser.java
package com.skocur.imagecipher.tools;

import javafx.scene.paint.Color;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.jetbrains.annotations.NotNull;

public class ColorParser {

  private static final Logger logger = LogManager.getLogger();

  public static int getColor(@NotNull Color color) {
    logger.debug("Parsing color: " + color.toString());
    int R = (int) Math.round(255 * color.getRed());
    int G = (int) Math.round(255 * color.getGreen());
    int B = (int) Math.round(255 * color.getBlue());

    R = (R << 16) & 0x00FF0000;
    G = (G << 8) & 0x0000FF00;
    B = B & 0x000000FF;

    return 0xFF000000 | R | G | B;
  }
}

File: app/src/main/kotlin/com/imagecipher/app/tools/ColorParser.kt
package com.imagecipher.app.tools

import javafx.scene.paint.Color
import kotlin.math.roundToInt

object ColorParser {

    fun getColor(color: Color): Int {
        var R = (255 * color.red).roundToInt()
        var G = (255 * color.green).roundToInt()
        var B = (255 * color.blue).roundToInt()

        R = (R shl 16) and 0x00FF0000
        G = (G shl 8) and 0x0000FF00
        B = B and 0x000000FF

        return 0xFF000000.toInt() or R or G or B
    }
}
